src/hunspell/hunspell.cxx

```diff
@@ -624,7 +624,8 @@ bool HunspellImpl::spell(const std::string& word, int* info, std::string* root)
         return false;
       return true;
     }
-    return true;
+
+    goto success;
```

Comment:
Instead of goto, could you extract the common part to a method/function, call that and then return?

To avoid repetitiveness where convert+return would occur multiple times, consider renaming existing method to an internal helper, and creating a wrapper which does the conversion.